valhalla/mjolnir/sqlite3.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <sqlite3.h>
+
+#include <optional>
+#include <string>
+
+namespace valhalla {
+namespace mjolnir {
+
+// RAII wrapper around sqlite3 and spatialite connection
+class Sqlite3 final {
+  sqlite3* db;
+  void* spatialite;
+
+  // constructor is private, use `open` instead
```

Comment:
but why though? i know the semantics of the underlying library have something similar named open, is that the only reason? to me this is extra complication/unexpected so im just wondering if there is a reason other than looking like the original library that we are wrapping

Reply:
No, the only reason for having the static method is not to throw an exception if object creation failed by some reason. Overall, I see 3 approaches here
- regular constructor + throwing an exception
- regular constructor + `Sqlite3` instance might be invalid (`db == null`) + bool operator to check that it is valid
- static method that returns `std::optional<Sqlite3>` so caller must check that object is valid

I decided to go with the last one for more obvious ergonomic.